apps/air-discount-scheme/backend/src/app/modules/user/user.service.ts

```diff
@@ -60,4 +90,17 @@ export class UserService {
   async getUserInfoByNationalId(nationalId: string): Promise<User | null> {
     return this.getUserByNationalId<User>(nationalId, User)
   }
+
+  async getMultipleUsersByNationalIdArray(
+    ids: string[],
+  ): Promise<Array<User | null>> {
+    const allUsers = ids.map(
+      async (nationalId) =>
+        (this.getUserInfoByNationalId(nationalId) as unknown) as User,
```

Comment:
it would be nice to get rid of the the `as` -- or simply filtering out nulls, since it looks like they can return null

so if you change the thing below to
```
return (await Promise.all(allUsers)).filter(Boolean)
```

you should get all the correct types without type cooercion (and it will handle these edge cases somewhat gracefully)

Reply:
also, you could make the return type `Array<User>` instead of having them possibly be null

Reply:
Just saw this now - Needed to refresh the page.